### DISCUSSION TITLE
Feature request - Adding a references class

### DISCUSSION THREAD

**Discussion by kulbhushanchand** [mmistakes/minimal-mistakes#2825] [Ideas]
When converting files from `RMarkdown `to `Markdown`, the references are tagged under  `references csl-bib-body hanging-indent` class. But this theme, does not have any class to nicely displayed any references. 

- Having a class to displays the refences would be a nice enhancement to the theme.
- Is there any other class pre-built into the theme, that I can use for the references?

Following is the generated section in `markdown` file.

```
<div id="refs" class="references csl-bib-body hanging-indent">

<div id="ref-baranowski2015" class="csl-entry">

Baranowski, Tom, Fran Blumberg, Richard Buday, Ann DeSmet, Lynn E.
Fiellin, C. Shawn Green, Pamela M. Kato, et al. 2015. “Games for Health
for Childrencurrent Status and Needed Research.” *Games for Health
Journal* 5 (1): 1–12. <https://doi.org/10.1089/g4h.2015.0026>.

</div>

<div id="ref-bifulco2017" class="csl-entry">

Bifulco, P., D. Esposito, G. D. Gargiulo, S. Savino, V. Niola, L.
Iuppariello, and M. Cesarelli. 2017. “2017 e-Health and Bioengineering
Conference (EHB).” In, 173–76.
<https://doi.org/10.1109/EHB.2017.7995389>.

</div>

</div>
```

**Comment by iBug** [comment]
I'd recommend instead that you use an ordered list for references just like this:

```markdown
### References

1. Blah blah blah
2. Blah blah blah
```

You can checkout [one of my articles][1] that uses this format. It looks decent to me.

Alternatively, you can make use of the Markdown footnote syntax `[^1]`, `[^2]` etc, which comes with Jekyll's default Markdown renderer Kramdown. See [this Stack Overflow answer][2] for details.

  [1]: https://ibug.io/cn/2019/06/about-internet-security/#references
  [2]: https://stackoverflow.com/a/48250535/5958455

---

The reason I'm suggesting this workaround is because how references look is mostly a personal preference and I'm unconfident there's a one-size-fits-all appearance that satisfies everyone. You can create your own look for `.references` if you have the basics of CSS, following [the documetation here][3].

  [3]: https://mmistakes.github.io/minimal-mistakes/docs/stylesheets/#customizing

**Reply by kulbhushanchand** [reply]
Thanks for the insights. 

I tried using lists and foot-notes earlier, which was time-consuming.
In my case the references are auto-generated when knitting the `rmd` files to `md` files.

```
a citation in the text  (Baranowski et al. 2015). 
More text.
Some more text.


## References 

<div id="refs" class="references csl-bib-body hanging-indent">

<div id="ref-baranowski2015" class="csl-entry">

Baranowski, Tom, Fran Blumberg, Richard Buday, Ann DeSmet, Lynn E.
Fiellin, C. Shawn Green, Pamela M. Kato, et al. 2015. “Games for Health
for Childrencurrent Status and Needed Research.” *Games for Health
Journal* 5 (1): 1–12. <https://doi.org/10.1089/g4h.2015.0026>.

</div>

</div>

```

Manually editing a number of citations to fit under any criteria would be very time-consuming and not a good approach. The `md` file is auto-generated and will over-write all the changes. Also, I'm using minimal mistakes remote theme, so probably tinkering with the core files may not be a good idea in terms of future updates.  

That is why, I suggested for enhancements to this theme i.e. a `reference class`. If provided in the theme, I just need to change the auto generated `reference class` with the `new reference class` in the theme.

**Reply by iBug** [reply]
You don't have to "tinker with the core files" to add your own CSS, even when using the remote theme method. You just copy `assets/css/main.scss` and add your stuff after everything else, [as described here][1].

To make your references look like a list, a bare minimal CSS addition would be as follows:

```css
.csl-entry {
  display: list-item;
}
```

[Customize list styles][2] as needed from here on.

  [1]: https://github.com/mmistakes/minimal-mistakes/discussions/2580#discussioncomment-178843
  [2]: https://www.w3schools.com/css/css_list.asp

**Reply by kulbhushanchand** [reply]
That was spot-on. I really appreciate your prompt help.

Following is the result at my local build - 
![image](https://user-images.githubusercontent.com/17742733/108092987-3f5d6e80-70a3-11eb-865a-4eedc4a65d24.png)
